File: backend/pkg/models/product.go
package models

type Product struct {
	Uid          string        `json:"uid,omitempty"`
	Id           string        `json:"id,omitempty"`
	Name         string        `json:"name,omitempty"`
	Price        float32       `json:"price,omitempty"`
	Transactions []Transaction `json:"was_bought,omitempty"`
	DType        []string      `json:"dgraph.type,omitempty"`
}

func NewProduct(id string, name string, price float32) (*Product, error) {
	return &Product{
		Id:    id,
		Name:  name,
		Price: price,
		DType: []string{"Product"},
	}, nil
}

File: transactions/app/src/main/java/com/example/transactions/models/Product.kt
package com.example.transactions.models

import com.google.gson.annotations.SerializedName

data class Product(
    @SerializedName("id")
    val id: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("price")
    val price: Int,
)
